src/lib.ml

```diff
@@ -996,6 +996,9 @@ module DB = struct
 
   type t = db
 
+  let equal = (==)
+  let hash = Hashtbl.hash
```

Comment:
Let's not use `Hashtbl.hash` here because db is mutable. Let's use some immutable part of it.

Reply:
Hmm, that's a bit hard actually. We need to resolve every library to do a proper comparison.

Reply:
Oh well, there is nothing to hash here. (a function, a hashtable and a lazy)
So if we want to use this as a part of the key, we have 2 reasonable choices:
- add a field `id`, and hash that
- don't hash this at all (equivalent to `let hash _ = 0`), and show that keys that differ only on db are rare. (assuming that's actually the case)

Reply:
Actually, all of these functions are unnecessary if we just take the super context as input. So for now we can punt on this.